File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/3235-check-if-the-rectangle-corner-is-reachable.js
var isReachable = function(targetX, targetY) {
    while (targetX > 1 || targetY > 1) {
        if (targetX > targetY) {
            if (targetY === 0) return false;
            targetX %= targetY;
        } else {
            if (targetX === 0) return false;
            targetY %= targetX;
        }
    }
    return true;
};

File: leetcode/AIcoding/3.DeepSeek-v3/erlang_solutions/3235-check-if-the-rectangle-corner-is-reachable.erl
-spec is_reachable(TargetX :: integer(), TargetY :: integer()) -> boolean().
is_reachable(TargetX, TargetY) ->
    GCD = fun
        GCD(0, B) -> B;
        GCD(A, B) -> GCD(B rem A, A)
    end,
    GCD(TargetX, TargetY) band 1 =:= 1.